lib/filterable_by.rb

```diff
@@ -29,7 +29,12 @@ def self.merge(scope, unscoped, hash, name, **opts, &block)
         sub = block.arity == 2 ? yield(unscoped, negative, **opts) : yield(negative, **opts)
         return nil unless sub
 
-        scope = scope.merge(sub.invert_where)
+        if sub.respond_to?(:invert_where)
+          sub = sub.invert_where
+        else
+          sub.where_clause = sub.where_clause.invert
+        end
+        scope = scope.merge(sub)
```

Comment:
I dont think it should affect us, but I did notice that `merge` behaviour has changed in Rails 7 https://guides.rubyonrails.org/7_0_release_notes.html#active-record-notable-changes.  would only be situation where we have same column referenced in multiple merged scopes.

Reply:
Interesting point, let me test something ...

Reply:
I added extra tests to confirm that filters cannot be used to remove "hard scopes", see https://github.com/bsm/filterable-by/pull/9/files#diff-6ff7d42eb62461e7e406d3c0773fb482d99c847c87414c0507536aabb90610ddR65